File: src/Shared/Logistics.Shared.Models/Payment/PaymentMethodDto.cs
using Logistics.Shared.Consts;

namespace Logistics.Shared.Models;

public record PaymentMethodDto
{
    public Guid Id { get; set; }
    public PaymentMethodType Type { get; set; }
    public PaymentMethodVerificationStatus VerificationStatus { get; set; }
    public bool IsDefault { get; set; }
    public required AddressDto BillingAddress { get; set; }

    // Card-specific
    public string? CardHolderName { get; set; }
    public string? CardNumber { get; set; } // Masked card number
    public string? Cvc { get; set; } // Masked CVC
    public int? ExpMonth { get; set; }
    public int? ExpYear { get; set; }

    // US Bank account-specific
    public string? AccountNumber { get; set; }
    public string? AccountHolderName { get; set; }
    public string? BankName { get; set; }
    public string? RoutingNumber { get; set; }
    public UsBankAccountHolderType AccountHolderType { get; set; }
    public UsBankAccountType? AccountType { get; set; }
    public string? VerificationUrl { get; set; } // Stripe verification URL for ACH verification

    // International Bank Account
    public string? SwiftCode { get; set; }
}

File: src/Client/Logistics.OfficeApp/src/app/core/api/models/payment/payment-method.dto.ts
import {AddressDto} from "../address.dto";
import {PaymentMethodType, UsBankAccountHolderType, UsBankAccountType} from "./enums";

export interface PaymentMethodDto {
  id: string;
  type: PaymentMethodType;
  isDefault: boolean;
  billingAddress: AddressDto;

  // Card-specific
  cardHolderName?: string;
  cardNumber?: string;
  cvc?: string;
  expMonth?: number;
  expYear?: number;

  // US Bank account-specific
  accountNumber?: string;
  accountHolderName?: string;
  bankName?: string;
  routingNumber?: string;
  accountHolderType?: UsBankAccountHolderType;
  accountType?: UsBankAccountType;

  // International Bank Account
  swiftCode?: string;
}
